pkg/testing/controller.go

```diff
@@ -84,3 +87,24 @@ func NewEstablishedCRD() *v1beta1.CustomResourceDefinition {
 		},
 	}
 }
+
+// AssertEventContains asserts that a k8s event stream contains a
+// value, and assert.FailNow() if it does not
+func AssertEventContains(t *gotesting.T, events <-chan string, contains string) {
+	select {
+	case e := <-events:
+		assert.Contains(t, e, contains)
+	case <-time.After(3 * time.Second):
```

Comment:
Should this duration be configurable?

Reply:
I've never had a need to change it. The event action is usually (always?) synchronous in testing, 3 seconds is really a "just in case" - it could probably be dropped lower.